packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -71,6 +75,28 @@ LocalizationsGenerator generateLocalizations({
   } on L10nException catch (e) {
     throwToolExit(e.message);
   }
+
+  final List<String> outputFileList = generator.outputFileList;
+  final File? untranslatedMessagesFile = generator.untranslatedMessagesFile;
+
+  // All other post processing.
+  if (options.format) {
+    final List<String> formatFileList = outputFileList.toList();
+    if (untranslatedMessagesFile != null) {
+      // Don't format the messages file using `dart format`.
+      formatFileList.remove(untranslatedMessagesFile.absolute.path);
+    }
+    if (formatFileList.isEmpty) {
+      return generator;
+    }
+    final String dartBinary = artifacts.getArtifactPath(Artifact.engineDartBinary);
+    final List<String> command = <String>[dartBinary, 'format', ...formatFileList];
+    final ProcessResult result = await processManager.run(command);
+    if (result.exitCode != 0) {
+      throwToolExit('Formatting failed: $result', exitCode: result.exitCode);
```

Comment:
how about showing the stdout and stderr of the `ProcessResult`?

Reply:
If formatting fails, is this expected to be user error or a tool bug? Because `throwToolExit` is generally reserved for the former--if this is the latter, we should throw a `ProcessException` or some such.

Reply:
yeah, assuming that the `generateLocalizations` generates the files properly, `dart format` should run properly so this isn't a user error. I've changed it to a `ProcessException`. Let me know if you're ok with the format I used for the message.

Reply:
made some minor textual suggestions, but otherwise lgtm